docs/design/ovs-pipeline.md

```diff
@@ -1,1197 +1,1941 @@
 
-When the Egress feature is enabled, extra flows will be added to
-[L3ForwardingTable], which send the egress traffic from Pods to external network
-to [SNATTable]. The following two flows match traffic to local Pods and traffic
-to the local Node IP respectively, and keep them in the normal forwarding path
-(to [L2ForwardingCalcTable]), so they will not be sent to [SNATTable]:
+Flows 1-2, matching packets with `EgressRuleCTLabel` set to 2, the `conj_id` allocated for the sample Kubernetes
+NetworkPolicy egress rule and loaded in table [EgressRule] flow 4, are utilized to collect metrics for the egress rule.
+
+Flows 3-4, matching packets with `EgressRuleCTLabel` set to 7, the `conj_id` allocated for the sample Antrea-native
+NetworkPolicy egress rule and loaded in table [EgressRule] flow 6, are utilized to collect metrics for the egress rule.
+
+Flow 5 serves as the default drop rule for the sample Antrea-native NetworkPolicy egress rule. It drops the packets with
+`APDenyRegMark` loaded in table [AntreaPolicyEgressRule] flow 9, and `APConjIDField` set to 5, the `conj_id` allocated
+the egress rule and loaded in table [AntreaPolicyEgressRule] flow 9.
+
+It's worth noting that ct label is used in flows 1-4, while reg is used in flow 5. The distinction lies in the fact that
+the value persisted in the ct label can be read throughout the entire lifecycle of a connection, but the reg mark is only valid for
+the current packet. For a connection permitted by a rule, all its packets should be collected for metrics, thus a ct
+label is used. For a connection denied or dropped by a rule, the first packet will be blocked, therefore a reg is enough.
+
+Flow 6 is the table-miss flow.
+
+### L3Forwarding
+
+This table, designated as the L3 routing table, serves to assign suitable source and destination MAC addresses to
+packets based on their destination IP addresses, as well as their reg marks or ct marks.
+
+If you dump the flows of this table, you may see the following:
 
 ```text
-table=70, priority=200,ip,reg0=0/0x80000,nw_dst=10.10.1.0/24 actions=goto_table:80
-table=70, priority=200,ip,reg0=0x2/0xffff,nw_dst=192.168.1.1 actions=goto_table:80
+1. table=L3Forwarding, priority=210,ip,nw_dst=10.10.0.1 actions=set_field:ba:5e:d1:55:aa:c0->eth_dst,set_field:0x20/0xf0->reg0,goto_table:L3DecTTL
+2. table=L3Forwarding, priority=210,ct_state=+rpl+trk,ct_mark=0x2/0xf,ip actions=set_field:ba:5e:d1:55:aa:c0->eth_dst,set_field:0x20/0xf0->reg0,goto_table:L3DecTTL
+3. table=L3Forwarding, priority=200,ip,reg0=0/0x200,nw_dst=10.10.0.0/24 actions=goto_table:L2ForwardingCalc
+4. table=L3Forwarding, priority=200,ip,nw_dst=10.10.1.0/24 actions=set_field:ba:5e:d1:55:aa:c0->eth_src,set_field:aa:bb:cc:dd:ee:ff->eth_dst,set_field:192.168.77.103->tun_dst,set_field:0x10/0xf0->reg0,goto_table:L3DecTTL
+5. table=L3Forwarding, priority=200,ip,reg0=0x200/0x200,nw_dst=10.10.0.24 actions=set_field:ba:5e:d1:55:aa:c0->eth_src,set_field:fa:b7:53:74:21:a6->eth_dst,goto_table:L3DecTTL
```

Comment:
I personally prefer reserving such flows, despite their potential redundancy, to maintain consistency throughout the document, since we have three test Pods in this document.

Reply:
@antoninbas Should we remove some redundancy flows that have the same pattern like above?